docs/API/helpers/components.md

```diff
@@ -0,0 +1,116 @@
+
+## 1. Install the basic software
+
+Install the [Fliplet CLI](/Quickstart.html) on your computer before continuing with the rest.
+
+---
+
+## 2. Generate the component boilerplate
+
+Once you have installed the [Fliplet CLI](/Quickstart.html), open the shell of your computer and run the `fliplet create-widget <package> <name> --helper` command described below to create the initial structure of the component in a new directory with the same name as the `<name>` in the input.
+
+The command requires two inputs:
+
+- **Package name**: the unique "reverse domain name" to be used for your component, e.g. `com.example.decision-tree`
+- **Component name**: the user-friendly name of your component, e.g. `Decision tree`
+
+```bash
+fliplet create-widget "com.fliplet.decision-tree" "Decision tree" --helper
+```
+
+Upon running the command above, you should receive an output similar to this:
+
+```
+Creating new widget Decision tree to C:\Users\John\decision-tree using helper boilerplate
+Component has been successfully created.
+```
+
+---
+
+## 3. Configure the component
+
+Before we starting copying over the helper code to the component, it's best to open up the `widget.json` file you find in the created folder and configure the component by reviewing and tweaking the following properties:
+
+- **`references`**
+  - This array property should list all [references](/components/Definition.html#references) between the fields used in your helper and Fliplet entities such as a **Data Source ID**, a **Screen ID** or a **Media File ID**. Furthermore, **views should be listed here as well** using the `richContent` reference type.
+
+    [References](/components/Definition.html#references) use the following structure: `<propertyName>:<referenceType>`
+
+    The following example assumes that you have a **view** field stored under the `content` property, a **dataSourceId** stored under the `contactsDataSource` property and a target action configured using the link provider, which has a **page** reference under `goTo`:
+
+    ```json
+    widget.json
+
+    {
+      "references": [
+        "content:richContent",
+        "contactsDataSource:dataSource",
+        "goTo.page:page"
+      ]
+    }
+    ```
+
+- **`dependencies` and `assets`**
+  - Under the `interface` and `build` top-level keys you'll find these two properties to be used to define first-party `dependencies` (Fliplet JS APIs or libraries) and `assets` defined by your helper codebase. See [dependencies and assets](/Dependencies-and-assets) and [component definition](/components/Definition.html#interface) for more information.
+
+---
+
+## 4. Copy the interface configuration
+
+Grab all contents of the `configuration` property of your helper JS code and paste it in the `js/interface.js` file of the component folder in the contents of the `Fliplet.Widget.generateInterface` function you will find in the boilerplate:
```

Comment:
```suggestion
Copy all contents of the `configuration` property of your helper JS code and paste it in the `js/interface.js` file of the component folder in the contents of the `Fliplet.Widget.generateInterface` function you will find in the boilerplate:
```

Reply:
Fixed in e76af57d2d0528e92b4fcc468a78edbcb80a2d65